jobmon_client/src/jobmon/client/task_group.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+from collections import defaultdict
+import warnings
+
+from typing import Dict, Iterable, Iterator, List, Optional, Set, Union
+from jobmon.client.task import Task
+
+
+def get_label_map(
+    tasks: Set[Task], labels: List[str]
+) -> Dict[str, Dict[Union[str, int], Set[Task]]]:
+    """Build up a mapping of tasks by label."""
+    label_map: Dict[str, Dict[Union[str, int], Set[Task]]] = defaultdict(
+        lambda: defaultdict(set)
+    )
+
```

Comment:
Parsing out these nested dictionary structures from raw code can be kind of difficult, a sample input in the docstring or in  a comment would be super useful. Just to show the expected structure
```
{
    'task_arg_1':
        {
            'val1': {t1, t2, t3}
        },
    'task_arg_2':
        {
            'val2': {t2, t3},
            'val3': {t4, t5}
        },
    'task_attribute':
        {
            'attr_val': {t1, t2}
        }
}
```

Reply:
that's a good point! although I didn't actually write this bit, Logan did. I'm not entirely positive what the structure of the map will be, although I can fiddle with it a little to make sure.